src/systems/joint_trajectory_controller/JointTrajectoryController.cc

```diff
@@ -0,0 +1,795 @@
+  {
+    this->target.position = _targetPoint.positions(_jointIndex);
+  }
+  if ((signed)_jointIndex < _targetPoint.velocities_size())
+  {
+    this->target.velocity = _targetPoint.velocities(_jointIndex);
+  }
+  if ((signed)_jointIndex < _targetPoint.accelerations_size())
+  {
+    this->target.acceleration = _targetPoint.accelerations(_jointIndex);
+  }
+  if ((signed)_jointIndex < _targetPoint.effort_size())
+  {
+    this->target.effort = _targetPoint.effort(_jointIndex);
+  }
+}
+
+void ActuatedJoint::Update(ignition::gazebo::EntityComponentManager &_ecm,
+                           const std::chrono::steady_clock::duration &_dt)
+{
+  // Get JointPosition and JointVelocity components
+  const auto jointPositionComponent = _ecm.Component<components::JointPosition>(this->entity);
+  const auto jointVelocityComponent = _ecm.Component<components::JointVelocity>(this->entity);
+
+  // Compute control errors and force for each PID controller
+  double forcePosition = 0.0, forceVelocity = 0.0;
+  if (!jointPositionComponent->Data().empty())
+  {
+    double errorPosition = jointPositionComponent->Data()[0] - this->target.position;
+    forcePosition = this->pids.position.Update(errorPosition, _dt);
+  }
+  if (!jointVelocityComponent->Data().empty())
+  {
+    double errorVelocity = jointVelocityComponent->Data()[0] - this->target.velocity;
+    forceVelocity = this->pids.velocity.Update(errorVelocity, _dt);
+  }
+
+  // Sum all forces
+  const double force = forcePosition + forceVelocity + this->target.effort;
+
+  // Get JointForceCmd component and apply command force
+  auto jointForceCmdComponent = _ecm.Component<components::JointForceCmd>(this->entity);
+  jointForceCmdComponent->Data()[0] = force;
+}
+
+void ActuatedJoint::ResetPIDs()
+{
+  this->pids.position.Reset();
+  this->pids.velocity.Reset();
+}
+
+//////////////////
+/// Trajectory ///
+//////////////////
+
+bool Trajectory::UpdateCurrentPoint(const std::chrono::steady_clock::duration &_simTime)
+{
+  bool isUpdated = false;
+
+  const auto trajectoryTime = _simTime - this->startTime;
+  while (true)
+  {
+    // Break if end of trajectory is reached (there are no more points after the current one)
+    if (this->IsGoalReached())
+    {
+      break;
+    }
+
+    // Break if point needs to be followed
+    const auto pointTFS = this->points[this->pointIndex].time_from_start();
+    const auto pointTime = std::chrono::seconds(pointTFS.sec()) +
+                           std::chrono::nanoseconds(pointTFS.nsec());
+    if (pointTime >= trajectoryTime)
+    {
+      break;
+    }
+
+    // Otherwise increment and try again (joint targets need to be updated)
+    ++this->pointIndex;
+    isUpdated = true;
+  };
+
+  // Return true if a point index was updated
+  return isUpdated;
+}
+
+bool Trajectory::IsGoalReached() const
+{
+  if (this->pointIndex + 1 >= this->points.size())
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
```

Comment:
Would this be preferred style-wise?
```suggestion
bool Trajectory::IsGoalReached() const
{
  return this->pointIndex + 1 >= this->points.size();
}
```

Reply:
+1, this looks more concise